src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -584,8 +600,34 @@ private void RemoveMetadataReferenceCore(VisualStudioMetadataReference reference
         /// </summary>
         private void OnImportChanged(object sender, EventArgs e)
         {
+            AssertIsForeground();
```

Comment:
easier would be to make this an "async void" method.  Then you can just call "await" below and not need to manually continuewith.

(though you would need a try/catch for cancellation exceptions).  so your call.